nixos/modules/services/monitoring/cockpit.nix

```diff
@@ -0,0 +1,186 @@
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.port ];
+
+    # units are in reverse sort order if you ls $out/lib/systemd/system
+    # all these units are basically verbatim translated from upstream
+    systemd.slices.system-cockpithttps = { # Translation from $out/lib/systemd/system/systemd-cockpithttps.slice
+      description = "Resource limits for all cockpit-ws-https@.service instances";
+      sliceConfig = {
+        TasksMax = 200;
+        MemoryHigh = "75%";
+        MemoryMax = "90%";
+      };
+    };
+    systemd.sockets."cockpit-wsinstance-https@" = { # Translation from $out/lib/systemd/system/cockpit-wsinstance-https@.socket
+      unitConfig = {
+        Description= "Socket for Cockpit Web Service https instance %I";
+        BindsTo= [ "cockpit.service" "cockpit-wsinstance-https@%i.service" ];
+        # clean up the socket after the service exits, to prevent fd leak
+        # this also effectively prevents a DoS by starting arbitrarily many sockets, as
+        # the services are resource-limited by system-cockpithttps.slice
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/https@%i.sock";
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+    systemd.services."cockpit-wsinstance-https@" = { # Translation from $out/lib/systemd/system/cockpit-wsinstance-https@.service
+      description = "Cockpit Web Service https instance %I";
+      stopIfChanged = true;
+      bindsTo = [ "cockpit.service"];
+      path = [ cfg.package ];
+      documentation = [ "man:cockpit-ws(8)" ];
+      serviceConfig = {
+        Slice = "system-cockpithttps.slice";
+        ExecStart = "${cfg.package}/libexec/cockpit-ws --for-tls-proxy --port=0";
+        User = "root";
+        Group = "";
+      };
+    };
+    systemd.sockets.cockpit-wsinstance-http = { # Translation from $out/lib/systemd/system/cockpit-wsinstance-http.socket
+      unitConfig = {
+        Description="Socket for Cockpit Web Service http instance";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/http.sock";
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+    systemd.sockets.cockpit-wsinstance-https-factory = { # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory.socket
+      unitConfig = {
+        Description="Socket for Cockpit Web Service https instance factory";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/https-factory.sock";
+        Accept=true;
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+    systemd.services."cockpit-wsinstance-https-factory@" = { # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory@.service
```

Comment:
```suggestion
    systemd.services."cockpit-wsinstance-https-factory@" = { 
```

Reply:
I added these comments for documentation reasons, if someone else will touch this code in the future

Reply:
I moved the comments to the line just above the unit to make the lines shorter